### ISSUE TITLE
PSR2 standard shouldn't require space after USE block when next code is a closing tag

### ISSUE THREAD

**Issue by evilebottnawi** [squizlabs/PHP_CodeSniffer#753]
I use code in template (html + php code)
```php
<?php
use insteria\feature\socialAuth\includes\Helper as Helper;
?>
<h1><?php echo Helper::socialTitle(); ?></h1>
```
For less code.
But got error: PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse
I think it is not normal

**Comment by aik099** [comment]
What is the error message text?

**Comment by evilebottnawi** [comment]
@aik099 
There must be one blank line after the last USE statement; 1 found;

**Comment by aik099** [comment]
What this sniff wants usually is correct because some other code goes after `use` statement. This particular case (the closing PHP tag is on the next line after `use` statement) I would consider as an exception and don't show that error.

**Comment by evilebottnawi** [comment]
@aik099 But in psr-2 space after use need only in class files. This is template, not class.

**Comment by aik099** [comment]
I'm not saying that you're wrong. I'm exactly saying (in my previous message), that your use case is absolutely valid and sniff needs to be adapted to handle that.

**Comment by evilebottnawi** [comment]
@aik099 may be add condition for check php tag in source file?

**Comment by aik099** [comment]
How would it help. You have `<?php` in both regular PHP files and php-template files. Anyway I believe @gsherwood already have enough context to fix the issue.

**Comment by gsherwood** [comment]
Technically, you need to write your code like this:
```php
<?php
use insteria\feature\socialAuth\includes\Helper as Helper;

?>
<h1><?php echo Helper::socialTitle(); ?></h1>
```
But that looks pretty silly. You or I could ask the PHP-FIG what their intention is for mixed PHP/HTML files like this, but we probably wouldn't get a good answer (it's doubtful it was even considered) and the standard wouldn't be changed anyway. So I've decided to just stop this error from being generated, even if it technically goes against PSR2.

**Comment by evilebottnawi** [comment]
@gsherwood thanks! I think it is right solution.
